### PR TITLE
NF: Replace gson with jackson

### PR DESCRIPTION
**Pull request by TarekkMA**
## Purpose / Description
Replace Gson with Jackson which is a more maintained and feature rich json parser

## Fixes
Related to #8668

## Approach
Gson was already there I just used Jackson the same as gson was used in the codebase

## How Has This Been Tested?
relaying on the existing tests

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
I'm okay with this - I believe the only downside is +400kilobyte APK weight, which is a shame but the performance increase was demonstrably large! Arthur's test collection will load something like 450% faster.

**Comment by TarekkMA** [comment]
yes, and other small loads may have from 200~350% increase.

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiSerialization.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ You should have received a copy of the GNU General Public License along with
+ this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.ichi2.anki;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public final class AnkiSerialization {
```
[the remark this answers is not in the document]
Comment (TarekkMA) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiSerialization.java:22]: Okay I will add javadocs to this class, the idea behind using once object mapper is to use same configuration throght the app, and it's mentioned [here](https://github.com/FasterXML/jackson-docs/wiki/Presentation:-Jackson-Performance) that `ObjectMapper` and `JsonFactory` are heavyweight .

as for why this is not in the original repo, I don't know exactly. maybe they don't like the idea of a singleton.

This class should be removed when AnkiDroid adopt a dependency injection/ service locator approach

**Comment by TarekkMA** [comment]
@Arthur-Milchior should I convert `org.json.JSON***` to use jackson?

I will try to convert one usage after the other each in a PR.

**Pull request merged**